store/etcd/rulesets_internal_test.go

```diff
@@ -189,3 +192,64 @@ func TestCompareSignature(t *testing.T) {
 		require.Equal(t, fmt.Sprintf("invalid %s with value '%s': %s", exp.Field, exp.Value, exp.Reason), exp.Error())
 	})
 }
+
+func BenchmarkProtoMarshalling(b *testing.B) {
+	rs, err := regula.NewBoolRuleset(
+		rule.New(rule.And(rule.Not(rule.BoolValue(false)), rule.BoolParam("param")), rule.BoolValue(true)),
+		rule.New(rule.And(rule.BoolParam("1st-param"), rule.BoolParam("2nd-param")), rule.BoolValue(false)),
+	)
+	require.NoError(b, err)
+
+	pbrs := toProtobufRuleset(rs)
```

Comment:
To have an accurate benchmark you need to change a few things:

- Move `toProtobufRuleset` in the loop: encoding a ruleset to protobuf means doing both actions, not just one
- Use `b.ResetTimer()` right before the loop

Reply:
Done in https://github.com/heetch/regula/pull/86/commits/bd1632056f5ee3d36049cc3b9fea6fd7ccf02681.